config

```diff
@@ -1,937 +1,854 @@
-	# 1. I'm most concerned about whether is a 7300LC is closer to a 7100 versus a 7100LC.
-	# 2. The variant 64-bit processors cause concern should GCC support explicit schedulers
-	#    for these chips in the future.
-	#         PA7300LC -> 7100LC (1.1)
-	#         PA8200   -> 8000   (2.0)
-	#         PA8500   -> 8000   (2.0)
-	#         PA8600   -> 8000   (2.0)
-
-	CPUSCHEDULE=`echo $CPUSCHEDULE|sed -e 's/7300LC/7100LC/' -e 's/8.00/8000/'`
-	# Finish Model transformations
-
-	__CNF_CPPFLAGS="$__CNF_CPPFLAGS -DB_ENDIAN"
-	__CNF_CFLAGS="$__CNF_CFLAGS -mschedule=$CPUSCHEDULE -march=$CPUARCH"
-	__CNF_CXXFLAGS="$__CNF_CXXFLAGS -mschedule=$CPUSCHEDULE -march=$CPUARCH"
-	OUT="linux-generic32" ;;
-  armv[1-3]*-*-linux2) OUT="linux-generic32" ;;
-  armv[7-9]*-*-linux2) OUT="linux-armv4"
-		       __CNF_CFLAGS="$__CNF_CFLAGS -march=armv7-a"
-		       __CNF_CXXFLAGS="$__CNF_CXXFLAGS -march=armv7-a"
-		       ;;
-  arm*-*-linux2) OUT="linux-armv4" ;;
-  aarch64-*-linux2) OUT="linux-aarch64" ;;
-  sh*b-*-linux2) OUT="linux-generic32";
-		 __CNF_CPPFLAGS="$__CNF_CPPFLAGS -DB_ENDIAN" ;;
-  sh*-*-linux2)	 OUT="linux-generic32";
-		 __CNF_CPPFLAGS="$__CNF_CPPFLAGS -DL_ENDIAN" ;;
-  m68k*-*-linux2) OUT="linux-generic32";
-		  __CNF_CPPFLAGS="$__CNF_CPPFLAGS -DB_ENDIAN" ;;
-  s390-*-linux2) OUT="linux-generic32";
-		 __CNF_CPPFLAGS="$__CNF_CPPFLAGS -DB_ENDIAN" ;;
-  s390x-*-linux2)
-	# To be uncommented when glibc bug is fixed, see Configure...
-	#if egrep -e '^features.* highgprs' /proc/cpuinfo >/dev/null ; then
-	#  echo "WARNING! If you wish to build \"highgprs\" 32-bit library, then you"
-	#  echo "         have to invoke './Configure linux32-s390x' *manually*."
-	#  if [ "$DRYRUN" = "false" -a -t -1 ]; then
-	#    echo "         You have about 5 seconds to press Ctrl-C to abort."
-	#    (trap "stty `stty -g`; exit 0" 2 0; stty -icanon min 0 time 50; read waste) <&1
-	#  fi
-	#fi
-	OUT="linux64-s390x"
-	;;
-  x86_64-*-linux?)
-	if $CC -dM -E -x c /dev/null 2>&1 | grep -q ILP32 > /dev/null; then
-	    OUT="linux-x32"
-	else
-	    OUT="linux-x86_64"
-	fi ;;
-  *86-*-linux2)
+my $MACHINE = $ENV{MACHINE} // &call_uname("-m");
+my $RELEASE = $ENV{RELEASE} // &call_uname("-r");
+my $SYSTEM = $ENV{SYSTEM} // &call_uname("-s");
+my $VERSION = &call_uname("-v");
+
+# Look for ISC/SCO with its unusual uname program
+sub is_sco_uname {
+    open UNAME, "uname -X 2>/dev/null|" or die "Can't pipe uname, $!";
+    my $line = "";
+    while ( <UNAME> ) {
+        chop;
+        $line = $_ if m@^Release@;
+    }
+    close UNAME;
+    return "" if $line eq '';
+    my @fields = split($line);
+    return $fields[2];
+}
+
+sub get_sco_type() {
+    my $REL = shift;
+
+    if ( -f "/etc/kconfig" ) {
+        return "${MACHINE}-whatever-isc4" if $REL eq '4.0' || $REL eq '4.1';
+    } else {
+        return "whatever-whatever-sco3" if $REL eq '3.2v4.2';
+        return "whatever-whatever-sco5" if $REL =~ m@3.2v5.0.*@;
```

Comment:
Don't forget to escape explicit periods.

Reply:
yes, fixed three places.